
## Categories

*As of Thursday, January 24 2019, we provide 488 software packages, in 7 categories, covering 41 fields of science:*

* [`viz`](#viz) <small>graphs, molecular visualization, plotting, remote display</small>
* [`system`](#system) <small></small>
* [`math`](#math) <small>computational geometry, deep learning, linear algebra, machine learning, numerical analysis, numerical library, optimization, scientific computing, statistics, symbolic</small>
* [`physics`](#physics) <small>astronomy, geophysics, geoscience, materials science, photonics</small>
* [`biology`](#biology) <small>clinical science, computational biology, cryo-em, genomics, neurology, phylogenetics</small>
* [`chemistry`](#chemistry) <small>cheminformatics, computational chemistry, cristallography, electrostatics, molecular dynamics, quantum chemistry</small>
* [`devel`](#devel) <small>build, compiler, data, data analytics, framework, language, lib, mpi, networking, parser</small>


!!! warning "Licensed software"

    Access to software modules marked with ^<b class="lic"></b>^ in the tables
    below is restricted to properly licensed user groups.

    The SRCC is not funded to provide commercial software on Sherlock and
    researchers are responsible for the costs of purchasing and renewing
    commercial software licenses. For more information, please feel free to
    [contact us](mailto:srcc-support@stanford.edu) and see the [Stanford
    Software Licensing page](https://uit.stanford.edu/service/softwarelic) for
    purchasing information.


<!-- color styles for module properties -->
<style>
.lic  { color: darkred; }
.lic:after {
    content: attr(class);
}
.soft_anch {
    display: inline-block;
    padding-top: 70px;
    margin-top: -70px;
}
</style>


### Viz
Field  | <img style="float:left;min-width:110px;visibility:hidden">Module&nbsp;name | <img style="float:left;min-width:90px;visibility:hidden">Version(s) | URL  | Description
:----- | :----------- | :------ | :--- | :----------
**graphs** | <a id="graphviz" class="soft_anch">`graphviz`</a> | -&nbsp;`2.40.1` | [Website](https://www.graphviz.org/) | Graphviz is open source graph visualization software. 
**molecular visualization** | <a id="pymol" class="soft_anch">`pymol`</a>^<b class="lic"></b>^ | -&nbsp;`1.8.6.2` | [Website](https://pymol.org) | PyMOL is a Python-enhanced molecular graphics tool. 
**plotting** | <a id="gnuplot" class="soft_anch">`gnuplot`</a> | -&nbsp;`5.2.0` | [Website](http://www.gnuplot.info) | Gnuplot is a portable command-line driven graphing utility for Linux, OS/2, MS Windows, OSX, VMS, and many other platforms. 
**plotting** | <a id="grace" class="soft_anch">`grace`</a> | -&nbsp;`5.1.25` | [Website](http://plasma-gate.weizmann.ac.il/Grace) | Grace is a WYSIWYG tool to make two-dimensional plots of numerical data. 
**plotting** | <a id="py-basemap" class="soft_anch">`py-basemap`</a> | -&nbsp;`1.1.0_py36`<br/>-&nbsp;`1.1.0_py27` | [Website](https://matplotlib.org/basemap/) | The matplotlib basemap toolkit is a library for plotting 2D data on maps in Python. 
**plotting** | <a id="py-matplotlib" class="soft_anch">`py-matplotlib`</a> | -&nbsp;`2.1.2_py27`<br/>-&nbsp;`2.1.2_py36`<br/>-&nbsp;`2.2.2_py36`<br/>-&nbsp;`2.2.2_py27` | [Website](http://matplotlib.org) | Matplotlib is a Python 2D plotting library which produces publication quality figures in a variety of hardcopy formats and interactive environments across platforms. 
**plotting** | <a id="py-plotly" class="soft_anch">`py-plotly`</a> | -&nbsp;`2.4.1_py27` | [Website](https://plot.ly/python/) | Plotly's Python graphing library makes interactive, publication-quality graphs online. 
**remote display** | <a id="virtualgl" class="soft_anch">`virtualgl`</a> | -&nbsp;`2.5.2` | [Website](http://www.virtualgl.org) | VirtualGL is an open source toolkit that gives any Unix or Linux remote display software the ability to run OpenGL applications with full 3D hardware acceleration. 

### System
Field  | <img style="float:left;min-width:110px;visibility:hidden">Module&nbsp;name | <img style="float:left;min-width:90px;visibility:hidden">Version(s) | URL  | Description
:----- | :----------- | :------ | :--- | :----------

### Math
Field  | <img style="float:left;min-width:110px;visibility:hidden">Module&nbsp;name | <img style="float:left;min-width:90px;visibility:hidden">Version(s) | URL  | Description
:----- | :----------- | :------ | :--- | :----------
**computational geometry** | <a id="cgal" class="soft_anch">`cgal`</a> | -&nbsp;`4.10` | [Website](http://www.cgal.org) | The Computational Geometry Algorithms Library (CGAL) is a C++ library that aims to provide easy access to efficient and reliable algorithms in computational geometry. 
**computational geometry** | <a id="qhull" class="soft_anch">`qhull`</a> | -&nbsp;`2015.2` | [Website](http://www.qhull.org/) | Qhull computes the convex hull, Delaunay triangulation, Voronoi diagram, halfspace intersection about a point, furthest-site Delaunay triangulation, and furthest-site Voronoi diagram. 
**deep learning** | <a id="caffe2" class="soft_anch">`caffe2`</a> | -&nbsp;`0.8.1` | [Website](https://caffe2.ai) | Caffe2 is a deep learning framework that provides an easy and straightforward way to experiment with deep learning and leverage community contributions of new models and algorithms. 
**deep learning** | <a id="cudnn" class="soft_anch">`cudnn`</a> | -&nbsp;`5.1`<br/>-&nbsp;`6.0`<br/>-&nbsp;`7.0.1`<br/>-&nbsp;`7.0.4`<br/>-&nbsp;`7.0.5`<br/>-&nbsp;`7.1.1`<br/>-&nbsp;`7.1.4`<br/>-&nbsp;`7.3.1.20`<br/>-&nbsp;`7.4.1.5` | [Website](https://developer.nvidia.com/cudnn) | NVIDIA cuDNN is a GPU-accelerated library of primitives for deep neural networks. 
**deep learning** | <a id="h2o" class="soft_anch">`h2o`</a> | -&nbsp;`3.22.0.1` | [Website](https://www.tensorflow.org) | TensorFlow™ is an open source software library for numerical computation using data flow graphs. 
**deep learning** | <a id="py-horovod" class="soft_anch">`py-horovod`</a> | -&nbsp;`0.12.1_py36`<br/>-&nbsp;`0.12.1_py27` | [Website](https://github.com/uber/horovod) | Horovod is a distributed training framework for TensorFlow. The goal of Horovod is to make distributed Deep Learning fast and easy to use. 
**deep learning** | <a id="py-keras" class="soft_anch">`py-keras`</a> | -&nbsp;`2.0.8_py27`<br/>-&nbsp;`2.1.5_py36`<br/>-&nbsp;`2.2.4_py27`<br/>-&nbsp;`2.2.4_py36`<br/>-&nbsp;`2.1.5_py27` | [Website](http://keras.io) | Keras is a high-level neural networks API, written in Python and capable of running on top of TensorFlow, CNTK, or Theano. 
**deep learning** | <a id="py-onnx" class="soft_anch">`py-onnx`</a> | -&nbsp;`1.0.1_py27` | [Website](https://onnx.ai) | ONNX is a open format to represent deep learning models. 
**deep learning** | <a id="py-pytorch" class="soft_anch">`py-pytorch`</a> | -&nbsp;`0.2.0_py27`<br/>-&nbsp;`0.2.0_py36`<br/>-&nbsp;`0.3.0_py36`<br/>-&nbsp;`1.0.0_py27`<br/>-&nbsp;`1.0.0_py36`<br/>-&nbsp;`0.3.0_py27` | [Website](http://pytorch.org) | PyTorch is a deep learning framework that puts Python first. 
**deep learning** | <a id="py-tensorflow" class="soft_anch">`py-tensorflow`</a> | -&nbsp;`1.4.0_py27`<br/>-&nbsp;`1.5.0_py27`<br/>-&nbsp;`1.5.0_py36`<br/>-&nbsp;`1.6.0_py36`<br/>-&nbsp;`1.7.0_py27`<br/>-&nbsp;`1.8.0_py27`<br/>-&nbsp;`1.9.0_py27`<br/>-&nbsp;`1.9.0_py36`<br/>-&nbsp;`1.12.0_py27`<br/>-&nbsp;`1.12.0_py36`<br/>-&nbsp;`1.6.0_py27` | [Website](https://www.tensorflow.org) | TensorFlow™ is an open source software library for numerical computation using data flow graphs. 
**deep learning** | <a id="py-tensorlayer" class="soft_anch">`py-tensorlayer`</a> | -&nbsp;`1.6.3_py27` | [Website](http://tensorlayer.org) | TensorLayer is a Deep Learning (DL) and Reinforcement Learning (RL) library extended from Google TensorFlow. 
**deep learning** | <a id="py-theano" class="soft_anch">`py-theano`</a> | -&nbsp;`1.0.1_py27` | [Website](http://deeplearning.net/software/theano) | Theano is a Python library that allows you to define, optimize, and evaluate mathematical expressions involving multi-dimensional arrays efficiently. 
**deep learning** | <a id="tensorrt" class="soft_anch">`tensorrt`</a> | -&nbsp;`3.0.1`<br/>-&nbsp;`3.0.4`<br/>-&nbsp;`4.0.1.6`<br/>-&nbsp;`5.0.2.6` | [Website](https://developer.nvidia.com/tensorrt) | NVIDIA TensorRT™ is a high-performance deep learning inference optimizer and runtime that delivers low latency, high-throughput inference for deep learning applications. 
**deep learning** | <a id="torch" class="soft_anch">`torch`</a> | -&nbsp;`20180202` | [Website](http://torch.ch) | Torch is a scientific computing framework with wide support for machine learning algorithms that puts GPUs first. 
**linear algebra** | <a id="armadillo" class="soft_anch">`armadillo`</a> | -&nbsp;`8.200.1` | [Website](http://arma.sourceforge.net/) | Armadillo is a high quality linear algebra library (matrix maths) for the C++ language, aiming towards a good balance between speed and ease of use. 
**machine learning** | <a id="py-scikit-learn" class="soft_anch">`py-scikit-learn`</a> | -&nbsp;`0.19.1_py36`<br/>-&nbsp;`0.19.1_py27` | [Website](http://scikit-learn.org) | Scikit-learn is a free software machine learning library for the Python programming language. 
**numerical analysis** | <a id="matlab" class="soft_anch">`matlab`</a>^<b class="lic"></b>^ | -&nbsp;`R2017a`<br/>-&nbsp;`R2017b`<br/>-&nbsp;`R2018a` | [Website](https://www.mathworks.com/products/matlab.html) | MATLAB (matrix laboratory) is a multi-paradigm numerical computing environment and fourth-generation programming language. 
**numerical analysis** | <a id="octave" class="soft_anch">`octave`</a> | -&nbsp;`4.2.1` | [Website](https://www.gnu.org/software/octave/) | GNU Octave is a high-level language primarily intended for numerical computations. 
**numerical library** | <a id="arpack" class="soft_anch">`arpack`</a> | -&nbsp;`3.5.0` | [Website](https://github.com/opencollab/arpack-ng) | Collection of Fortran77 subroutines designed to solve large scale eigenvalue problems. 
**numerical library** | <a id="fftw" class="soft_anch">`fftw`</a> | -&nbsp;`3.3.6` | [Website](http://www.fftw.org/) | The Fastest Fourier Transform in the West (FFTW) is a software library for computing discrete Fourier transforms (DFTs). 
**numerical library** | <a id="glpk" class="soft_anch">`glpk`</a> | -&nbsp;`4.63` | [Website](http://www.openblas.net/) | The GLPK (GNU Linear Programming Kit) package is intended for solving large-scale linear programming (LP), mixed integer programming (MIP), and other related problems. 
**numerical library** | <a id="gmp" class="soft_anch">`gmp`</a> | -&nbsp;`6.1.2` | [Website](https://gmplib.org) | GMP is a free library for arbitrary precision arithmetic, operating on signed integers, rational numbers, and floating-point numbers. 
**numerical library** | <a id="gsl" class="soft_anch">`gsl`</a> | -&nbsp;`1.16`<br/>-&nbsp;`2.3` | [Website](https://www.gnu.org/software/gsl) | The GNU Scientific Library (GSL) is a numerical library for C and C++ programmers. The library provides a wide range of mathematical routines such as random number generators, special functions and least-squares fitting. 
**numerical library** | <a id="harminv" class="soft_anch">`harminv`</a> | -&nbsp;`1.4.1` | [Website](https://github.com/stevengj/harminv) | harminv is a program designed to solve the problem of harmonic inversion: given a time series consisting of a sum of sinusoids (modes), extract their frequencies and amplitudes. 
**numerical library** | <a id="imkl" class="soft_anch">`imkl`</a> | -&nbsp;`2017.u2`<br/>-&nbsp;`2018.u1`<br/>-&nbsp;`2018`<br/>-&nbsp;`2019` | [Website](https://software.intel.com/en-us/intel-mkl) | Intel Math Kernel Library (Intel MKL) is a library of optimized math routines for science, engineering, and financial applications. Core math functions include BLAS, LAPACK, ScaLAPACK, sparse solvers, fast Fourier transforms, and vector math.[3] The routines in MKL are hand-optimized specifically for Intel processors 
**numerical library** | <a id="libxsmm" class="soft_anch">`libxsmm`</a> | -&nbsp;`1.8.1` | [Website](https://github.com/hfp/libxsmm) | LIBXSMM is a library for small dense and small sparse matrix-matrix multiplications as well as for deep learning primitives such as small convolutions 
**numerical library** | <a id="metis" class="soft_anch">`metis`</a> | -&nbsp;`5.1.0` | [Website](http://glaros.dtc.umn.edu/gkhome/metis/metis/overview) | METIS is a set of serial programs for partitioning graphs, partitioning finite element meshes, and producing fill reducing orderings for sparse matrices. 
**numerical library** | <a id="mpfr" class="soft_anch">`mpfr`</a> | -&nbsp;`3.1.5` | [Website](https://www.mpfr.org) | The MPFR library is a C library for multiple-precision floating-point computations with correct rounding. 
**numerical library** | <a id="mumps" class="soft_anch">`mumps`</a> | -&nbsp;`5.1.2` | [Website](http://mumps.enseeiht.fr/) | A parallel sparse direct solver. 
**numerical library** | <a id="nagcl" class="soft_anch">`nagcl`</a> | -&nbsp;`cll6i26dcl` | [Website](https://www.nag.com/nag-c-library) | The NAG C Library is the largest and most comprehensive collection of mathematical and statistical algorithms for C and C++. 
**numerical library** | <a id="nagfl" class="soft_anch">`nagfl`</a> | -&nbsp;`fll6i26dcl` | [Website](https://www.nag.com/nag-fortran-library) | The NAG Fortran Library is the largest and most comprehensive collection of numerical and statistical algorithms in Fortran. 
**numerical library** | <a id="nagfs" class="soft_anch">`nagfs`</a> | -&nbsp;`fsl6i26dcl` | [Website](https://www.nag.com/nag-library-smp-multicore) | The NAG Library for SMP & Multicore is based on, and includes, the full functionality of the NAG Fortran Library. 
**numerical library** | <a id="nagmb" class="soft_anch">`nagmb`</a> | -&nbsp;`MBL6I25DNL` | [Website](https://www.nag.com/nag-toolbox-matlab) | The NAG C Library is the largest and most comprehensive collection of mathematical and statistical algorithms for C and C++. 
**numerical library** | <a id="openblas" class="soft_anch">`openblas`</a> | -&nbsp;`0.2.19`<br/>-&nbsp;`0.3.4` | [Website](http://www.openblas.net/) | OpenBLAS is an optimized BLAS library 
**numerical library** | <a id="parmetis" class="soft_anch">`parmetis`</a> | -&nbsp;`4.0.3` | [Website](http://glaros.dtc.umn.edu/gkhome/metis/parmetis/overview) | ParMETIS is an MPI-based parallel library that implements a variety of algorithms for partitioning unstructured graphs, meshes, and for computing fill-reducing orderings of sparse matrices. 
**numerical library** | <a id="py-numpy" class="soft_anch">`py-numpy`</a> | -&nbsp;`1.14.3_py36`<br/>-&nbsp;`1.14.3_py27` | [Website](http://www.numpy.org) | NumPy is the fundamental package for scientific computing with Python. 
**numerical library** | <a id="py-pyublas" class="soft_anch">`py-pyublas`</a> | -&nbsp;`2017.1_py27` | [Website](https://mathema.tician.de/software/pyublas/) | PyUblas provides a seamless glue layer between Numpy and Boost.Ublas for use with Boost.Python. 
**numerical library** | <a id="py-scipy" class="soft_anch">`py-scipy`</a> | -&nbsp;`1.1.0_py36`<br/>-&nbsp;`1.1.0_py27` | [Website](https://www.scipy.org/scipylib) | The SciPy library provides many user-friendly and efficient numerical routines such as routines for numerical integration and optimization. 
**numerical library** | <a id="qrupdate" class="soft_anch">`qrupdate`</a> | -&nbsp;`1.1.2` | [Website](https://sourceforge.net/projects/qrupdate/) | qrupdate is a Fortran library for fast updates of QR and Cholesky decompositions. 
**numerical library** | <a id="scalapack" class="soft_anch">`scalapack`</a> | -&nbsp;`2.0.2` | [Website](http://www.netlib.org/scalapack/) | ScaLAPACK is a library of high-performance linear algebra routines for parallel distributed memory machines. 
**numerical library** | <a id="scotch" class="soft_anch">`scotch`</a> | -&nbsp;`6.0.4` | [Website](http://www.labri.fr/perso/pelegrin/scotch) | Software package and libraries for sequential and parallel graph partitioning, static mapping and clustering, sequential mesh and hypergraph partitioning, and sequential and parallel sparse matrix block ordering. 
**numerical library** | <a id="superlu" class="soft_anch">`superlu`</a> | -&nbsp;`5.2.1` | [Website](https://github.com/xiaoyeli/superlu) | SuperLU is a general purpose library for the direct solution of large, sparse, nonsymmetric systems of linear equations. 
**numerical library** | <a id="xblas" class="soft_anch">`xblas`</a> | -&nbsp;`1.0.248` | [Website](http://www.netlib.org/xblas) | Extra precise basic linear algebra subroutines. 
**optimization** | <a id="gurobi" class="soft_anch">`gurobi`</a> | -&nbsp;`7.5.1`<br/>-&nbsp;`8.0.1_py27`<br/>-&nbsp;`8.0.1_py36` | [Website](http://www.gurobi.com) | The Gurobi Optimizer is a commercial optimization solver for mathematical programming. 
**optimization** | <a id="knitro" class="soft_anch">`knitro`</a> | -&nbsp;`10.3.0` | [Website](https://www.artelys.com/en/optimization-tools/knitro) | Artelys Knitro is an optimization solver for difficult large-scale nonlinear problems. 
**scientific computing** | <a id="py-scipystack" class="soft_anch">`py-scipystack`</a> | -&nbsp;`1.0_py36`<br/>-&nbsp;`1.0_py27` | [Website](https://www.scipy.org) | The SciPy Stack is a collection of open source software for scientific computing in Python. It provides the following packages: numpy, scipy, matplotlib, ipython, jupyter, pandas, sympy and nose. 
**statistics** | <a id="R" class="soft_anch">`R`</a> | -&nbsp;`3.4.0`<br/>-&nbsp;`3.5.1` | [Website](http://r-project.org) | R is a free software environment for statistical computing and graphics. <br/>[Usage on Sherlock](https://www.sherlock.stanford.edu/docs/software/using/R)
**statistics** | <a id="datamash" class="soft_anch">`datamash`</a> | -&nbsp;`1.3` | [Website](https://www.gnu.org/software/datamash/) | GNU datamash is a command-line program which performs basic numeric, textual and statistical operations on input textual data files. 
**statistics** | <a id="jags" class="soft_anch">`jags`</a> | -&nbsp;`4.3.0` | [Website](http://mcmc-jags.sourceforge.net/) | Just another Gibbs sampler (JAGS) is a program for simulation from Bayesian hierarchical models using Markov chain Monte Carlo (MCMC). 
**statistics** | <a id="py-rpy2" class="soft_anch">`py-rpy2`</a> | -&nbsp;`2.8.6_py27`<br/>-&nbsp;`2.9.2_py36` | [Website](https://rpy2.bitbucket.io) | rpy2 is an interface to R running embedded in a Python process. 
**statistics** | <a id="r-rstan" class="soft_anch">`r-rstan`</a> | -&nbsp;`2.17.3` | [Website](http://mc-stan.org/users/interfaces/rstan) | RStan is the R interface to Stan, an open-source software for facilitating statistical inference at the frontiers of applied statistics. 
**statistics** | <a id="rstudio" class="soft_anch">`rstudio`</a> | -&nbsp;`1.1.423` | [Website](http://www.rstudio.com) | RStudio is an integrated development environment (IDE) for R. It includes a console, syntax-highlighting editor that supports direct code execution, as well as tools for plotting, history, debugging and workspace management. 
**statistics** | <a id="sas" class="soft_anch">`sas`</a>^<b class="lic"></b>^ | -&nbsp;`9.4` | [Website](http://www.sas.com/) | SAS is a software suite developed by SAS Institute for advanced analytics, multivariate analyses, business intelligence, data management, and predictive analytics. 
**statistics** | <a id="stata" class="soft_anch">`stata`</a>^<b class="lic"></b>^ | -&nbsp;`14`<br/>-&nbsp;`15` | [Website](https://www.stata.com) | Stata is a complete, integrated statistical software package that provides everything you need for data analysis, data management, and graphics. 
**symbolic** | <a id="libmatheval" class="soft_anch">`libmatheval`</a> | -&nbsp;`1.1.11` | [Website](https://www.gnu.org/software/libmatheval/) | GNU libmatheval is a library (callable from C and Fortran) to parse and evaluate symbolic expressions input as text. 
**symbolic** | <a id="py-sympy" class="soft_anch">`py-sympy`</a> | -&nbsp;`1.1.1_py27`<br/>-&nbsp;`1.1.1_py36` | [Website](http://www.sympy.org) | SymPy is a Python library for symbolic mathematics. 

### Physics
Field  | <img style="float:left;min-width:110px;visibility:hidden">Module&nbsp;name | <img style="float:left;min-width:90px;visibility:hidden">Version(s) | URL  | Description
:----- | :----------- | :------ | :--- | :----------
**astronomy** | <a id="heasoft" class="soft_anch">`heasoft`</a> | -&nbsp;`6.22.1` | [Website](https://heasarc.gsfc.nasa.gov/docs/software/lheasoft/) | HEAsoft is a Unified Release of the FTOOLS (General and mission-specific tools to manipulate FITS files) and XANADU (High-level, multi-mission tasks for X-ray astronomical spectral, timing, and imaging data analysis) software packages. 
**geophysics** | <a id="opensees" class="soft_anch">`opensees`</a> | -&nbsp;`2.5.0` | [Website](http://opensees.berkeley.edu/) | OpenSees is a software framework for developing applications to simulate the performance of structural and geotechnical systems subjected to earthquakes. 
**geoscience** | <a id="gdal" class="soft_anch">`gdal`</a> | -&nbsp;`2.2.1` | [Website](http://www.gdal.org) | GDAL is a translator library for raster and vector geospatial data formats. 
**geoscience** | <a id="geos" class="soft_anch">`geos`</a> | -&nbsp;`3.6.2` | [Website](https://trac.osgeo.org/geos) | GEOS (Geometry Engine - Open Source) is a C++ port of Java Topology Suite (JTS). 
**geoscience** | <a id="proj" class="soft_anch">`proj`</a> | -&nbsp;`4.9.3` | [Website](http://proj4.org) | proj.4 is a standard UNIX filter function which converts geographic longitude and latitude coordinates into cartesian coordinates (and vice versa. 
**geoscience** | <a id="py-opendrift" class="soft_anch">`py-opendrift`</a> | -&nbsp;`1.0.3_py27` | [Website](https://github.com/OpenDrift/opendrift) | OpenDrift is a software for modeling the trajectories and fate of objects or substances drifting in the ocean, or even in the atmosphere. 
**geoscience** | <a id="py-pyproj" class="soft_anch">`py-pyproj`</a> | -&nbsp;`1.9.5.1_py36`<br/>-&nbsp;`1.9.5.1_py27` | [Website](https://github.com/jswhit/pyproj) | Python interface to PROJ4 library for cartographic transformations. 
**geoscience** | <a id="udunits" class="soft_anch">`udunits`</a> | -&nbsp;`2.2.26` | [Website](https://www.unidata.ucar.edu/software/udunits/) | The UDUNITS package from Unidata is a C-based package for the programatic handling of units of physical quantities. 
**materials science** | <a id="atat" class="soft_anch">`atat`</a> | -&nbsp;`3.36` | [Website](https://www.brown.edu/Departments/Engineering/Labs/avdw/atat/) | Alloy Theoretic Automated Toolkit: a software toolkit for modeling coupled configurational and vibrational disorder in alloy systems. 
**photonics** | <a id="meep" class="soft_anch">`meep`</a> | -&nbsp;`1.3`<br/>-&nbsp;`1.4.3` | [Website](https://github.com/stevengj/meep) | Meep is a free finite-difference time-domain (FDTD) simulation software package to model electromagnetic systems. 
**photonics** | <a id="mpb" class="soft_anch">`mpb`</a> | -&nbsp;`1.5`<br/>-&nbsp;`1.6.2` | [Website](https://github.com/stevengj/mpb) | MPB is a free software package for computing the band structures, or dispersion relations, and electromagnetic modes of periodic dielectric structures, on both serial and parallel computers. 

### Biology
Field  | <img style="float:left;min-width:110px;visibility:hidden">Module&nbsp;name | <img style="float:left;min-width:90px;visibility:hidden">Version(s) | URL  | Description
:----- | :----------- | :------ | :--- | :----------
**clinical science** | <a id="simvascular" class="soft_anch">`simvascular`</a> | -&nbsp;`20180704` | [Website](https://simvascular.github.io) | Simvascular is a blood flow simulation and analysis toolkit. This module provides the svFSI (Fluid Solid Interaction) solver. 
**computational biology** | <a id="imp" class="soft_anch">`imp`</a> | -&nbsp;`2.8.0` | [Website](https://integrativemodeling.org/) | IMP's broad goal is to contribute to a comprehensive structural characterization of biomolecules ranging in size and complexity from small peptides to large macromolecular assemblies, by integrating data from diverse biochemical and biophysical experiments. 
**computational biology** | <a id="py-biopython" class="soft_anch">`py-biopython`</a> | -&nbsp;`1.70_py27` | [Website](http://biopython.org) | Biopython is a set of freely available tools for biological computation written in Python. 
**computational biology** | <a id="rosetta" class="soft_anch">`rosetta`</a> | -&nbsp;`3.8` | [Website](https://www.rosettacommons.org) | Rosetta is the premier software suite for modeling macromolecular structures. As a flexible, multi-purpose application, it includes tools for structure prediction, design, and remodeling of proteins and nucleic acids. 
**cryo-em** | <a id="eman2" class="soft_anch">`eman2`</a> | -&nbsp;`2.2` | [Website](http://blake.bcm.edu/emanwiki/EMAN2) | EMAN2 is a broadly based greyscale scientific image processing suite with a primary focus on processing data from transmission electron microscopes. 
**cryo-em** | <a id="relion" class="soft_anch">`relion`</a> | -&nbsp;`2.0.3`<br/>-&nbsp;`2.1` | [Website](http://www2.mrc-lmb.cam.ac.uk/relion/index.php/Main_Page) | RELION (for REgularised LIkelihood OptimisatioN, pronounce rely-on) is a stand-alone computer program that employs an empirical Bayesian approach to refinement of (multiple) 3D reconstructions or 2D class averages in electron cryo-microscopy (cryo-EM). 
**genomics** | <a id="angsd" class="soft_anch">`angsd`</a> | -&nbsp;`0.919` | [Website](http://www.popgen.dk/angsd/index.php/ANGSD) | ANGSD is a software for analyzing next generation sequencing data. 
**genomics** | <a id="bcftools" class="soft_anch">`bcftools`</a> | -&nbsp;`1.6`<br/>-&nbsp;`1.8` | [Website](https://github.com/samtools/bcftools) | BCFtools is a program for variant calling and manipulating files in the Variant Call Format (VCF) and its binary counterpart BCF. 
**genomics** | <a id="bcl2fastq" class="soft_anch">`bcl2fastq`</a> | -&nbsp;`2.20` | [Website](https://support.illumina.com/downloads/bcl2fastq-conversion-software-v2-20.html) | The bcl2fastq2 conversion software can be used to convert BCL files from MiniSeq, MiSeq, NextSeq, HiSeq, iSeq and NovaSeq sequening systems. 
**genomics** | <a id="bedtools" class="soft_anch">`bedtools`</a> | -&nbsp;`2.27.1` | [Website](https://bedtools.readthedocs.io) | The bedtools utilities are a swiss-army knife of tools for a wide-range of genomics analysis tasks. 
**genomics** | <a id="bowtie" class="soft_anch">`bowtie`</a> | -&nbsp;`1.2.2` | [Website](http://bowtie-bio.sourceforge.net/) | Bowtie is an ultrafast, memory-efficient short read aligner. 
**genomics** | <a id="bowtie2" class="soft_anch">`bowtie2`</a> | -&nbsp;`2.3.4.1` | [Website](http://bowtie-bio.sourceforge.net/bowtie2) | Bowtie 2 is an ultrafast and memory-efficient tool for aligning sequencing reads to long reference sequences. 
**genomics** | <a id="bwa" class="soft_anch">`bwa`</a> | -&nbsp;`0.7.17` | [Website](http://bio-bwa.sourceforge.net/) | BWA (Burrows-Wheeler Aligner) is a software package for mapping low-divergent sequences against a large reference genome, such as the human genome. 
**genomics** | <a id="cufflinks" class="soft_anch">`cufflinks`</a> | -&nbsp;`2.2.1` | [Website](https://cole-trapnell-lab.github.io/cufflinks/) | Cufflinks assembles transcripts, estimates their abundances, and tests for differential expression and regulation in RNA-Seq samples. 
**genomics** | <a id="fastqc" class="soft_anch">`fastqc`</a> | -&nbsp;`0.11.8` | [Website](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) | FastQC aims to provide a simple way to do some quality control checks on raw sequence data coming from high throughput sequencing pipelines. 
**genomics** | <a id="fastx_toolkit" class="soft_anch">`fastx_toolkit`</a> | -&nbsp;`0.0.14` | [Website](http://hannonlab.cshl.edu/fastx_toolkit/) | The FASTX-Toolkit is a collection of command line tools for Short-Reads FASTA/FASTQ files preprocessing. 
**genomics** | <a id="freebayes" class="soft_anch">`freebayes`</a> | -&nbsp;`1.2.0` | [Website](https://github.com/ekg/freebayes) | FreeBayes is a Bayesian genetic variant detector designed to find small polymorphisms. 
**genomics** | <a id="hic-pro" class="soft_anch">`hic-pro`</a> | -&nbsp;`2.10.0` | [Website](https://github.com/nservant/HiC-Pro) | HiC-Pro: An optimized and flexible pipeline for Hi-C data processing. 
**genomics** | <a id="hisat2" class="soft_anch">`hisat2`</a> | -&nbsp;`2.1.0` | [Website](http://ccb.jhu.edu/software/hisat2) | HISAT2 is a fast and sensitive alignment program for mapping next-generation sequencing reads (both DNA and RNA) to a population of human genomes (as well as to a single reference genome). 
**genomics** | <a id="htslib" class="soft_anch">`htslib`</a> | -&nbsp;`1.6`<br/>-&nbsp;`1.8` | [Website](https://github.com/samtools/htslib) | C library for high-throughput sequencing data formats. 
**genomics** | <a id="jellyfish" class="soft_anch">`jellyfish`</a> | -&nbsp;`2.2.10` | [Website](https://github.com/gmarcais/Jellyfish) | A fast multi-threaded k-mer counter. 
**genomics** | <a id="kallisto" class="soft_anch">`kallisto`</a> | -&nbsp;`0.44.0` | [Website](https://pachterlab.github.io/kallisto/) | kallisto is a program for quantifying abundances of transcripts from RNA-Seq data using high-throughput sequencing reads. 
**genomics** | <a id="mixcr" class="soft_anch">`mixcr`</a> | -&nbsp;`2.1.12` | [Website](https://milaboratory.com/software/mixcr/) | MiXCR is a universal framework that processes big immunome data from raw sequences to quantitated clonotypes. 
**genomics** | <a id="ncbi-blast+" class="soft_anch">`ncbi-blast+`</a> | -&nbsp;`2.6.0`<br/>-&nbsp;`2.7.1` | [Website](https://blast.ncbi.nlm.nih.gov/Blast.cgi) | NCBI BLAST+ is a suite of command-line tools to run BLAST (Basic Local Alignment Search Tool), an algorithm for comparing primary biological sequence information. 
**genomics** | <a id="plink" class="soft_anch">`plink`</a> | -&nbsp;`1.07`<br/>-&nbsp;`1.90b5.3`<br/>-&nbsp;`2.0a1` | [Website](http://zzz.bwh.harvard.edu/plink/) | PLINK is a free, open-source whole genome association analysis toolset, designed to perform a range of basic, large-scale analyses in a computationally efficient manner. 
**genomics** | <a id="py-bx-python" class="soft_anch">`py-bx-python`</a> | -&nbsp;`0.8.1_py27` | [Website](https://github.com/bxlab/bx-python) | Tools for manipulating biological data, particularly multiple sequence alignments. 
**genomics** | <a id="py-cutadapt" class="soft_anch">`py-cutadapt`</a> | -&nbsp;`1.18_py36`<br/>-&nbsp;`1.18_py27` | [Website](https://cutadapt.readthedocs.io) | Cutadapt finds and removes adapter sequences, primers, poly-A tails and other types of unwanted sequence from your high-throughput sequencing reads. 
**genomics** | <a id="py-fithic" class="soft_anch">`py-fithic`</a> | -&nbsp;`1.1.3_py27` | [Website](https://github.com/ay-lab/fithic) | Fit-Hi-C is a tool for assigning statistical confidence estimates to chromosomal contact maps produced by genome architecture assays. 
**genomics** | <a id="py-macs2" class="soft_anch">`py-macs2`</a> | -&nbsp;`2.1.1_py27` | [Website](https://github.com/taoliu/MACS) | MACS (Model-based Analysis of ChIP-Seq) implements a novel ChIP-Seq analysis method. 
**genomics** | <a id="py-multiqc" class="soft_anch">`py-multiqc`</a> | -&nbsp;`1.6_py36`<br/>-&nbsp;`1.6_py27` | [Website](https://multiqc.info) | MultiQC is a reporting tool that parses summary statistics from results and log files generated by other bioinformatics tools. 
**genomics** | <a id="py-pysam" class="soft_anch">`py-pysam`</a> | -&nbsp;`0.14.1_py27` | [Website](https://github.com/pysam-developers/pysam) | Pysam is a python module for reading, manipulating and writing genomic data sets. 
**genomics** | <a id="salmon" class="soft_anch">`salmon`</a> | -&nbsp;`0.12.0` | [Website](https://combine-lab.github.io/salmon) | Highly-accurate & wicked fast transcript-level quantification from RNA-seq reads using lightweight alignments. 
**genomics** | <a id="samtools" class="soft_anch">`samtools`</a> | -&nbsp;`1.6`<br/>-&nbsp;`1.8` | [Website](https://github.com/samtools/samtools) | Tools (written in C using htslib) for manipulating next-generation sequencing data. 
**genomics** | <a id="sentieon" class="soft_anch">`sentieon`</a>^<b class="lic"></b>^ | -&nbsp;`201808.01` | [Website](https://www.sentieon.com/) | Sentieon Genomics software is a set of software tools that perform analysis of genomic data obtained from DNA sequencing. 
**genomics** | <a id="star" class="soft_anch">`star`</a> | -&nbsp;`2.5.4b` | [Website](https://github.com/alexdobin/STAR) | STAR: ultrafast universal RNA-seq aligner. 
**genomics** | <a id="tophat" class="soft_anch">`tophat`</a> | -&nbsp;`2.1.1` | [Website](http://ccb.jhu.edu/software/tophat/index.shtml) | TopHat is a fast splice junction mapper for RNA-Seq reads. 
**genomics** | <a id="trim_galore" class="soft_anch">`trim_galore`</a> | -&nbsp;`0.5.0` | [Website](https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/) | Trim Galore! is a wrapper script to automate quality and adapter trimming as well as quality control, with some added functionality to remove biased methylation positions for RRBS sequence files. 
**genomics** | <a id="trinity" class="soft_anch">`trinity`</a> | -&nbsp;`2.8.4` | [Website](http://trinityrnaseq.github.io) | Trinity RNA-Seq de novo transcriptome assembly. 
**genomics** | <a id="vcflib" class="soft_anch">`vcflib`</a> | -&nbsp;`1.0.0` | [Website](https://github.com/vcflib/vcflib) | A C++ library for parsing and manipulating VCF files. 
**genomics** | <a id="vcftools" class="soft_anch">`vcftools`</a> | -&nbsp;`0.1.15` | [Website](https://vcftools.github.io) | VCFtools is a program package designed for working with VCF files, such as those generated by the 1000 Genomes Project. 
**neurology** | <a id="afni" class="soft_anch">`afni`</a> | -&nbsp;`17.2.07`<br/>-&nbsp;`18.2.04` | [Website](https://afni.nimh.nih.gov) | AFNI (Analysis of Functional NeuroImages) is a set of C programs for processing, analyzing, and displaying functional MRI (FMRI) data - a technique for mapping human brain activity. 
**neurology** | <a id="ants" class="soft_anch">`ants`</a> | -&nbsp;`2.1.0` | [Website](https://stnava.github.io/ANTs) | ANTs computes high-dimensional mappings to capture the statistics of brain structure and function. 
**neurology** | <a id="dcm2niix" class="soft_anch">`dcm2niix`</a> | -&nbsp;`1.0.20171215` | [Website](https://github.com/rordenlab/dcm2niix) | dcm2niix is a program esigned to convert neuroimaging data from the DICOM format to the NIfTI format. 
**neurology** | <a id="freesurfer" class="soft_anch">`freesurfer`</a> | -&nbsp;`6.0.0` | [Website](https://surfer.nmr.mgh.harvard.edu/) | An open source software suite for processing and analyzing (human) brain MRI images. 
**neurology** | <a id="fsl" class="soft_anch">`fsl`</a> | -&nbsp;`5.0.10` | [Website](https://fsl.fmrib.ox.ac.uk/fsl) | FSL is a comprehensive library of analysis tools for FMRI, MRI and DTI brain imaging data. 
**neurology** | <a id="mricron" class="soft_anch">`mricron`</a> | -&nbsp;`20160502` | [Website](https://www.nitrc.org/projects/mricron) | MRIcron is a cross-platform NIfTI format image viewer. 
**neurology** | <a id="mrtrix" class="soft_anch">`mrtrix`</a> | -&nbsp;`0.3.16` | [Website](http://www.mrtrix.org) | MRtrix3 provides a set of tools to perform various types of diffusion MRI analyses, from various forms of tractography through to next-generation group-level analyses. 
**neurology** | <a id="py-mdt" class="soft_anch">`py-mdt`</a> | -&nbsp;`0.10.9_py36` | [Website](https://github.com/cbclab/MDT) | The Maastricht Diffusion Toolbox, MDT, is a framework and library for parallelized (GPU and multi-core CPU) diffusion Magnetic Resonance Imaging (MRI) modeling. 
**neurology** | <a id="py-nipype" class="soft_anch">`py-nipype`</a> | -&nbsp;`1.1.3_py27`<br/>-&nbsp;`1.1.3_py36` | [Website](https://nipype.readthedocs.io/en/latest/index.html) | Nipype is a Python project that provides a uniform interface to existing neuroimaging software and facilitates interaction between these packages within a single workflow. 
**neurology** | <a id="spm" class="soft_anch">`spm`</a> | -&nbsp;`12` | [Website](http://www.fil.ion.ucl.ac.uk/spm/) | The SPM software package has been designed for the analysis of brain imaging data sequences. The sequences can be a series of images from different cohorts, or time-series from the same subject. 
**neurology** | <a id="workbench" class="soft_anch">`workbench`</a> | -&nbsp;`1.3.1` | [Website](https://www.humanconnectome.org/software/connectome-workbench) | Connectome Workbench is an open source, freely available visualization and discovery tool used to map neuroimaging data, especially data generated by the Human Connectome Project. 
**phylogenetics** | <a id="py-ete" class="soft_anch">`py-ete`</a> | -&nbsp;`3.0.0_py27` | [Website](http://etetoolkit.org/) | A Python framework for the analysis and visualization of trees. 

### Chemistry
Field  | <img style="float:left;min-width:110px;visibility:hidden">Module&nbsp;name | <img style="float:left;min-width:90px;visibility:hidden">Version(s) | URL  | Description
:----- | :----------- | :------ | :--- | :----------
**cheminformatics** | <a id="py-rdkit" class="soft_anch">`py-rdkit`</a> | -&nbsp;`2018.09.1_py36`<br/>-&nbsp;`2018.09.1_py27` | [Website](http://www.rdkit.org) | RDKit is a collection of cheminformatics and machine-learning software written in C++ and Python. 
**computational chemistry** | <a id="gaussian" class="soft_anch">`gaussian`</a>^<b class="lic"></b>^ | -&nbsp;`g16` | [Website](https://www.gaussian.com) | Gaussian is a general purpose computational chemistry software package. 
**computational chemistry** | <a id="libint" class="soft_anch">`libint`</a> | -&nbsp;`1.1.4`<br/>-&nbsp;`2.0.3` | [Website](https://sourceforge.net/p/libint/home/) | Libint computes molecular integrals. 
**computational chemistry** | <a id="libxc" class="soft_anch">`libxc`</a> | -&nbsp;`3.0.0` | [Website](http://octopus-code.org/wiki/Libxc) | Libxc is a library of exchange-correlation functionals for density-functional theory. 
**computational chemistry** | <a id="nwchem" class="soft_anch">`nwchem`</a> | -&nbsp;`6.8` | [Website](http://www.nwchem-sw.org) | NWChem is an ab initio computational chemistry software package which also includes quantum chemical and molecular dynamics functionality. 
**computational chemistry** | <a id="py-ase" class="soft_anch">`py-ase`</a> | -&nbsp;`3.14.1_py27` | [Website](https://wiki.fysik.dtu.dk/ase/) | The Atomic Simulation Environment (ASE) is a set of tools and Python modules for setting up, manipulating, running, visualizing and analyzing atomistic simulations. 
**computational chemistry** | <a id="schrodinger" class="soft_anch">`schrodinger`</a>^<b class="lic"></b>^ | -&nbsp;`2017-3`<br/>-&nbsp;`2018-1`<br/>-&nbsp;`2018-2` | [Website](https://www.schrodinger.com) | Schrödinger Suites (Small-molecule Drug Discovery Suite, Material Science Suite, Biologics Suite) provide a set of molecular modelling software. 
**computational chemistry** | <a id="vasp" class="soft_anch">`vasp`</a>^<b class="lic"></b>^ | -&nbsp;`5.4.1` | [Website](https://www.vasp.at) | The Vienna Ab initio Simulation Package (VASP) is a computer program for atomic scale materials modelling, e.g. electronic structure calculations and quantum-mechanical molecular dynamics, from first principles. 
**cristallography** | <a id="vesta" class="soft_anch">`vesta`</a> | -&nbsp;`3.4.4` | [Website](http://jp-minerals.org/vesta/en/) | VESTA is a 3D visualization program for structural models, volumetric data such as electron/nuclear densities, and crystal morphologies. 
**electrostatics** | <a id="apbs" class="soft_anch">`apbs`</a> | -&nbsp;`1.5` | [Website](http://www.poissonboltzmann.org/) | APBS solves the equations of continuum electrostatics for large biomolecular assemblages. 
**molecular dynamics** | <a id="gromacs" class="soft_anch">`gromacs`</a> | -&nbsp;`2016.3`<br/>-&nbsp;`2018` | [Website](http://www.gromacs.org) | GROMACS is a versatile package to perform molecular dynamics, i.e. simulate the Newtonian equations of motion for systems with hundreds to millions of particles. 
**molecular dynamics** | <a id="lammps" class="soft_anch">`lammps`</a> | -&nbsp;`20180316` | [Website](http://lammps.sandia.gov) | LAMMPS is a classical molecular dynamics code that models an ensemble of particles in a liquid, solid, or gaseous state. 
**molecular dynamics** | <a id="openmm" class="soft_anch">`openmm`</a> | -&nbsp;`7.1.1` | [Website](http://openmm.org) | A high performance toolkit for molecular simulation. 
**molecular dynamics** | <a id="plumed" class="soft_anch">`plumed`</a> | -&nbsp;`2.3.2` | [Website](http://www.plumed.org/) | PLUMED is an open source library for free energy calculations in molecular systems. 
**molecular dynamics** | <a id="py-raspa2" class="soft_anch">`py-raspa2`</a> | -&nbsp;`2.0.3_py27` | [Website](https://github.com/numat/RASPA2) | RASPA2 is a general purpose classical simulation package that can be used for the simulation of molecules in gases, fluids, zeolites, aluminosilicates, metal-organic frameworks, carbon nanotubes and external fields. 
**molecular dynamics** | <a id="qbox" class="soft_anch">`qbox`</a> | -&nbsp;`1.65.0` | [Website](http://www.qboxcode.org) | Qbox is a First-Principles Molecular Dynamics code. 
**molecular dynamics** | <a id="quip" class="soft_anch">`quip`</a> | -&nbsp;`20170901` | [Website](https://github.com/libAtoms/QUIP) | The QUIP package is a collection of software tools to carry out molecular dynamics simulations. 
**quantum chemistry** | <a id="cp2k" class="soft_anch">`cp2k`</a> | -&nbsp;`4.1` | [Website](http://www.cp2k.org) | CP2K is a quantum chemistry and solid state physics software package that can perform atomistic simulations of solid state, liquid, molecular, periodic, material, crystal, and biological systems. 
**quantum chemistry** | <a id="quantum-espresso" class="soft_anch">`quantum-espresso`</a> | -&nbsp;`6.2.1` | [Website](https://www.quantum-espresso.org) | Quantum ESPRESSO is an integrated suite of Open-Source computer codes for electronic-structure calculations and materials modeling at the nanoscale. It is based on density-functional theory, plane waves, and pseudopotentials. 
**quantum chemistry** | <a id="quantum-espresso_gpu" class="soft_anch">`quantum-espresso_gpu`</a> | -&nbsp;`1.1` | [Website](https://github.com/fspiga/qe-gpu) | GPU-accelerated Quantum ESPRESSO using CUDA FORTRAN 

### Devel
Field  | <img style="float:left;min-width:110px;visibility:hidden">Module&nbsp;name | <img style="float:left;min-width:90px;visibility:hidden">Version(s) | URL  | Description
:----- | :----------- | :------ | :--- | :----------
**build** | <a id="bazel" class="soft_anch">`bazel`</a> | -&nbsp;`0.16.1` | [Website](https://bazel.build/) | Bazel is a fast, scalable, multi-language and extensible build system. 
**build** | <a id="cmake" class="soft_anch">`cmake`</a> | -&nbsp;`3.8.1`<br/>-&nbsp;`3.11.1`<br/>-&nbsp;`3.13.1` | [Website](https://www.cmake.org) | CMake is an extensible, open-source system that manages the build process in an operating system and in a compiler-independent manner. 
**build** | <a id="scons" class="soft_anch">`scons`</a> | -&nbsp;`2.5.1_py36`<br/>-&nbsp;`2.5.1_py27` | [Website](http://www.scons.org) | SCons is an Open Source software construction tool. 
**compiler** | <a id="gcc" class="soft_anch">`gcc`</a> | -&nbsp;`7.1.0`<br/>-&nbsp;`7.3.0`<br/>-&nbsp;`8.1.0`<br/>-&nbsp;`6.3.0` | [Website](http://gcc.gnu.org) | The GNU Compiler Collection includes front ends for C, C++, Fortran, Java, and Go, as well as libraries for these languages (libstdc++, libgcj,...). 
**compiler** | <a id="icc" class="soft_anch">`icc`</a> | -&nbsp;`2017.u2`<br/>-&nbsp;`2018.u1`<br/>-&nbsp;`2018`<br/>-&nbsp;`2019` | [Website](https://software.intel.com/en-us/c-compilers) | Intel C++ Compiler, also known as icc or icl, is a group of C and C++ compilers from Intel 
**compiler** | <a id="ifort" class="soft_anch">`ifort`</a> | -&nbsp;`2017.u2`<br/>-&nbsp;`2018.u1`<br/>-&nbsp;`2018`<br/>-&nbsp;`2019` | [Website](https://software.intel.com/en-us/fortran-compilers) | Intel Fortran Compiler, also known as ifort, is a group of Fortran compilers from Intel 
**compiler** | <a id="llvm" class="soft_anch">`llvm`</a> | -&nbsp;`3.8.1`<br/>-&nbsp;`4.0.0`<br/>-&nbsp;`5.0.0`<br/>-&nbsp;`7.0.0` | [Website](http://llvm.org) | The LLVM Project is a collection of modular and reusable compiler and toolchain technologies. Clang is an LLVM native C/C++/Objective-C compiler, 
**compiler** | <a id="nagfor" class="soft_anch">`nagfor`</a> | -&nbsp;`npl6a61na` | [Website](https://www.nag.com/nag-compiler) | The NAG Fortran Compiler is a full standard implementation of the ISO Fortran 95 language with the addition of all of Fortran 2003, most of Fortran 2008 and OpenMP 3.0 and 3.1. 
**compiler** | <a id="pgi" class="soft_anch">`pgi`</a> | -&nbsp;`17.4`<br/>-&nbsp;`18.4` | [Website](http://pgroup.com) | PGI compilers and tools, including Open MPI (Community Edition). 
**compiler** | <a id="smlnj" class="soft_anch">`smlnj`</a> | -&nbsp;`110.81` | [Website](http://smlnj.org/) | Standard ML of New Jersey (abbreviated SML/NJ) is a compiler for the Standard ML '97 programming language. 
**data** | <a id="h5utils" class="soft_anch">`h5utils`</a> | -&nbsp;`1.12.1` | [Website](http://ab-initio.mit.edu/wiki/index.php/H5utils) | h5utils is a set of utilities for visualization and conversion of scientific data in the free, portable HDF5 format. 
**data** | <a id="hdf5" class="soft_anch">`hdf5`</a> | -&nbsp;`1.10.0p1`<br/>-&nbsp;`1.10.2` | [Website](https://www.hdfgroup.org/hdf5) | HDF5 is a data model, library, and file format for storing and managing data. It supports an unlimited variety of datatypes, and is designed for flexible and efficient I/O and for high volume and complex data. 
**data** | <a id="hiredis" class="soft_anch">`hiredis`</a> | -&nbsp;`0.13.3` | [Website](https://github.com/redis/hiredis) | Hiredis is a minimalistic C client library for the Redis database. 
**data** | <a id="ncl" class="soft_anch">`ncl`</a> | -&nbsp;`6.4.0` | [Website](http://www.ncl.ucar.edu/) | NCL is a free interpreted language designed specifically for scientific data processing and visualization. 
**data** | <a id="netcdf" class="soft_anch">`netcdf`</a> | -&nbsp;`4.4.1.1` | [Website](https://www.unidata.ucar.edu/software/netcdf) | NetCDF is a set of software libraries and self-describing, machine-independent data formats that support the creation, access, and sharing of array-oriented scientific data. 
**data** | <a id="pnetcdf" class="soft_anch">`pnetcdf`</a> | -&nbsp;`1.8.1` | [Website](http://cucis.ece.northwestern.edu/projects/PnetCDF/) | Parallel netCDF (PnetCDF) is a parallel I/O library for accessing NetCDF files in CDF-1, 2, and 5 formats. 
**data** | <a id="protobuf" class="soft_anch">`protobuf`</a> | -&nbsp;`3.4.0` | [Website](https://developers.google.com/protocol-buffers/) | Protocol Buffers (a.k.a., protobuf) are Google's language-neutral, platform-neutral, extensible mechanism for serializing structured data. 
**data** | <a id="py-pandas" class="soft_anch">`py-pandas`</a> | -&nbsp;`0.23.0_py27`<br/>-&nbsp;`0.23.0_py36` | [Website](https://pandas.pydata.org/) | pandas is an open source, BSD-licensed library providing high-performance, easy-to-use data structures and data analysis tools for the Python programming language. 
**data** | <a id="py-protobuf" class="soft_anch">`py-protobuf`</a> | -&nbsp;`3.4.0_py36`<br/>-&nbsp;`3.6.1_py27`<br/>-&nbsp;`3.6.1_py36`<br/>-&nbsp;`3.4.0_py27` | [Website](https://developers.google.com/protocol-buffers) | Python bindings for Google's Protocol Buffers data interchange format. 
**data** | <a id="redis" class="soft_anch">`redis`</a> | -&nbsp;`4.0.1` | [Website](https://redis.io) | Redis is an open source, in-memory data structure store, used as a database, cache and message broker. 
**data analytics** | <a id="hadoop" class="soft_anch">`hadoop`</a> | -&nbsp;`3.1.0` | [Website](https://hadoop.apache.org/) | The Apache Hadoop software library is a framework that allows for the distributed processing of large data sets across clusters of computers using simple programming models. 
**data analytics** | <a id="py-sparkhpc" class="soft_anch">`py-sparkhpc`</a> | -&nbsp;`0.3` | [Website](https://sparkhpc.readthedocs.io) | Launching and controlling spark on HPC clusters  
**data analytics** | <a id="spark" class="soft_anch">`spark`</a> | -&nbsp;`2.3.0` | [Website](https://spark.apache.org) | Apache Spark™ is a unified analytics engine for large-scale data processing. <br/>[Usage on Sherlock](https://www.sherlock.stanford.edu/docs/software/using/spark)
**framework** | <a id="dotnet" class="soft_anch">`dotnet`</a> | -&nbsp;`2.1.500` | [Website](https://dotnet.microsoft.com/) | .NET is a free, cross-platform, open source developer platform for building many different types of applications. 
**language** | <a id="cuda" class="soft_anch">`cuda`</a> | -&nbsp;`8.0.61`<br/>-&nbsp;`9.1.85`<br/>-&nbsp;`9.2.148`<br/>-&nbsp;`10.0.130`<br/>-&nbsp;`9.0.176` | [Website](https://developer.nvidia.com/cuda-toolkit) | CUDA is a parallel computing platform and application programming interface (API) model created by Nvidia. It allows software developers and software engineers to use a CUDA-enabled graphics processing unit (GPU) for general purpose processing. 
**language** | <a id="go" class="soft_anch">`go`</a> | -&nbsp;`1.9` | [Website](http://golang.org) | Go is an open source programming language that makes it easy to build simple, reliable, and efficient software. 
**language** | <a id="guile" class="soft_anch">`guile`</a> | -&nbsp;`2.0.11`<br/>-&nbsp;`2.2.2` | [Website](https://www.gnu.org/software/guile/) | GNU Guile is the preferred extension system for the GNU Project, which features an implementation of the Scheme programming language. 
**language** | <a id="java" class="soft_anch">`java`</a> | -&nbsp;`1.8.0_131` | [Website](http://www.oracle.com/technetwork/java/index.html) | Java is a general-purpose computer programming language that is concurrent, class-based, object-oriented,[14] and specifically designed to have as few implementation dependencies as possible. 
**language** | <a id="julia" class="soft_anch">`julia`</a> | -&nbsp;`0.6.4`<br/>-&nbsp;`1.0.0` | [Website](https://julialang.org) | Julia is a high-level, high-performance dynamic programming language for numerical computing. <br/>[Usage on Sherlock](https://www.sherlock.stanford.edu/docs/software/using/julia)
**language** | <a id="lua" class="soft_anch">`lua`</a> | -&nbsp;`5.3.4` | [Website](https://www.lua.org) | Lua is a powerful, efficient, lightweight, embeddable scripting language. It supports procedural programming, object-oriented programming, functional programming, data-driven programming, and data description. 
**language** | <a id="luarocks" class="soft_anch">`luarocks`</a> | -&nbsp;`2.4.3` | [Website](https://luarocks.org/) | LuaRocks is the package manager for Lua modules. 
**language** | <a id="manticore" class="soft_anch">`manticore`</a> | -&nbsp;`20180301` | [Website](http://manticore.cs.uchicago.edu/) | Manticore is a high-level parallel programming language aimed at general-purpose applications running on multi-core processors. 
**language** | <a id="nodejs" class="soft_anch">`nodejs`</a> | -&nbsp;`8.9.4`<br/>-&nbsp;`9.5.0` | [Website](https://nodejs.org/) | Node.js is a JavaScript runtime built on Chrome's V8 JavaScript engine. It provides the npm package manager. 
**language** | <a id="perl" class="soft_anch">`perl`</a> | -&nbsp;`5.26.0` | [Website](https://www.perl.org) | Perl 5 is a highly capable, feature-rich programming language with over 29 years of development. <br/>[Usage on Sherlock](https://www.sherlock.stanford.edu/docs/software/using/perl)
**language** | <a id="php" class="soft_anch">`php`</a> | -&nbsp;`7.3.0` | [Website](http://www.php.net) | PHP (recursive acronym for PHP: Hypertext Preprocessor) is an open source general-purpose scripting language that is especially suited for web development. 
**language** | <a id="py-cython" class="soft_anch">`py-cython`</a> | -&nbsp;`0.27.3_py36`<br/>-&nbsp;`0.27.3_py27` | [Website](http://cython.org/) | Cython is an optimising static compiler for both the Python programming language and the extended Cython programming language (based on Pyrex). 
**language** | <a id="py-ipython" class="soft_anch">`py-ipython`</a> | -&nbsp;`6.1.0_py36`<br/>-&nbsp;`5.4.1_py27` | [Website](https://ipython.org) | IPython is a command shell for interactive computing in multiple programming languages, originally developed for the Python programming language. 
**language** | <a id="py-jupyter" class="soft_anch">`py-jupyter`</a> | -&nbsp;`1.0.0_py36`<br/>-&nbsp;`1.0.0_py27` | [Website](https://jupyter.org) | Jupyter is a browser-based interactive notebook for programming, mathematics, and data science. It supports a number of languages via plugins. 
**language** | <a id="python" class="soft_anch">`python`</a> | -&nbsp;`3.6.1`<br/>-&nbsp;`2.7.13` | [Website](https://www.python.org) | Python is an interpreted, interactive, object-oriented programming language. <br/>[Usage on Sherlock](https://www.sherlock.stanford.edu/docs/software/using/python)
**language** | <a id="ruby" class="soft_anch">`ruby`</a> | -&nbsp;`2.4.1` | [Website](https://www.ruby-lang.org) | A dynamic, open source programming language with a focus on simplicity and productivity. It has an elegant syntax that is natural to read and easy to write. 
**language** | <a id="scala" class="soft_anch">`scala`</a> | -&nbsp;`2.12.6` | [Website](https://www.scala-lang.org) | Scala combines object-oriented and functional programming in one concise, high-level language. 
**lib** | <a id="ant" class="soft_anch">`ant`</a> | -&nbsp;`1.10.1` | [Website](https://ant.apache.org/) | Apache Ant is a Java library and command-line tool whose mission is to drive processes described in build files as targets and extension points dependent upon each other. 
**lib** | <a id="boost" class="soft_anch">`boost`</a> | -&nbsp;`1.64.0`<br/>-&nbsp;`1.69.0` | [Website](http://www.boost.org) | Boost is a set of libraries for the C++ programming language that provide support for tasks and structures such as linear algebra, pseudorandom number generation, multithreading, image processing, regular expressions, and unit testing. 
**lib** | <a id="cnmem" class="soft_anch">`cnmem`</a> | -&nbsp;`1.0.0` | [Website](https://github.com/NVIDIA/ncmem) | CNMeM is a simple library to help the Deep Learning frameworks manage CUDA memory. 
**lib** | <a id="cub" class="soft_anch">`cub`</a> | -&nbsp;`1.7.3` | [Website](https://nvlabs.github.io/cub/) | CUB is a flexible library of cooperative threadblock primitives and other utilities for CUDA kernel programming. 
**lib** | <a id="cutlass" class="soft_anch">`cutlass`</a> | -&nbsp;`0.1.0` | [Website](https://github.com/NVIDIA/cutlass) | CUTLASS is a collection of CUDA C++ template abstractions for implementing high-performance matrix-multiplication (GEMM) at all levels and scales within CUDA. 
**lib** | <a id="eigen" class="soft_anch">`eigen`</a> | -&nbsp;`3.3.3` | [Website](http://eigen.tuxfamily.org) | Eigen is a C++ template library for linear algebra: matrices, vectors, numerical solvers, and related algorithms. 
**lib** | <a id="libctl" class="soft_anch">`libctl`</a> | -&nbsp;`3.2.2`<br/>-&nbsp;`4.0.1` | [Website](https://github.com/stevengj/libctl) | libctl is a library for supporting flexible control files in scientific simulations. 
**lib** | <a id="libgpuarray" class="soft_anch">`libgpuarray`</a> | -&nbsp;`0.7.5` | [Website](http://deeplearning.net/software/libgpuarray) | Library to manipulate tensors on the GPU. 
**lib** | <a id="nccl" class="soft_anch">`nccl`</a> | -&nbsp;`1.3.4`<br/>-&nbsp;`2.0.4`<br/>-&nbsp;`2.1.15`<br/>-&nbsp;`2.2.13`<br/>-&nbsp;`2.3.7` | [Website](https://github.com/NVIDIA/nccl) | NCCL (pronounced 'Nickel') is a stand-alone library of standard collective communication routines, such as all-gather, reduce, broadcast, etc., that have been optimized to achieve high bandwidth over PCIe. 
**lib** | <a id="opencv" class="soft_anch">`opencv`</a> | -&nbsp;`3.3.0` | [Website](https://opencv.org) | OpenCV (Open Source Computer Vision Library) is an open source computer vision and machine learning software library. 
**lib** | <a id="py-h5py" class="soft_anch">`py-h5py`</a> | -&nbsp;`2.8.0_py36`<br/>-&nbsp;`2.7.1_py27` | [Website](http://www.h5py.org) | The h5py package is a Pythonic interface to the HDF5 binary data format. 
**lib** | <a id="py-netcdf4" class="soft_anch">`py-netcdf4`</a> | -&nbsp;`1.3.1_py36`<br/>-&nbsp;`1.3.1_py27` | [Website](https://unidata.github.io/netcdf4-python/) | netcdf4-python is a Python interface to the netCDF C library. 
**lib** | <a id="py-numba" class="soft_anch">`py-numba`</a> | -&nbsp;`0.35.0_py36`<br/>-&nbsp;`0.35.0_py27` | [Website](http://numba.pydata.org) | Numba is a compiler for Python array and numerical functions that gives you the power to speed up your applications with high performance functions written directly in Python.. 
**lib** | <a id="py-pycuda" class="soft_anch">`py-pycuda`</a> | -&nbsp;`2017.1.1_py27` | [Website](https://mathema.tician.de/software/pycuda/) | PyCUDA lets you access Nvidia‘s CUDA parallel computation API from Python. 
**lib** | <a id="py-scikit-image" class="soft_anch">`py-scikit-image`</a> | -&nbsp;`0.13.0_py27`<br/>-&nbsp;`0.14.0_py27` | [Website](http://scikit-image.org/) | scikit-image is a collection of algorithms for image processing. 
**lib** | <a id="swig" class="soft_anch">`swig`</a> | -&nbsp;`3.0.12` | [Website](http://www.swig.org) | SWIG is an interface compiler that connects programs written in C and C++ with scripting languages such as Perl, Python, Ruby, and Tcl. 
**lib** | <a id="tbb" class="soft_anch">`tbb`</a> | -&nbsp;`2017.u2`<br/>-&nbsp;`2018.u1`<br/>-&nbsp;`2018`<br/>-&nbsp;`2019` | [Website](https://software.intel.com/en-us/intel-tbb) | Intel® Threading Building Blocks (Intel® TBB) is a widely used C++ library for shared-memory parallel programming and heterogeneous computing (intra-node distributed memory programming). 
**lib** | <a id="trilinos" class="soft_anch">`trilinos`</a> | -&nbsp;`12.12.1` | [Website](https://trilinos.org/) | Trilinos is a collection of open-source software libraries, called packages, intended to be used as building blocks for the development of scientific applications. 
**lib** | <a id="zeromq" class="soft_anch">`zeromq`</a> | -&nbsp;`4.2.2` | [Website](http://zeromq.org) | ZeroMQ (also spelled ØMQ, 0MQ or ZMQ) is a high-performance asynchronous messaging library, aimed at use in distributed or concurrent applications. 
**mpi** | <a id="impi" class="soft_anch">`impi`</a> | -&nbsp;`2017.u2`<br/>-&nbsp;`2018.u1`<br/>-&nbsp;`2018`<br/>-&nbsp;`2019` | [Website](https://software.intel.com/en-us/intel-mpi-library) | Intel® MPI Library is a multi-fabric message passing library that implements the Message Passing Interface, version 3.1 (MPI-3.1) specification. 
**mpi** | <a id="openmpi" class="soft_anch">`openmpi`</a> | -&nbsp;`2.0.2`<br/>-&nbsp;`2.1.1`<br/>-&nbsp;`3.0.1`<br/>-&nbsp;`4.0.0`<br/>-&nbsp;`3.1.2` | [Website](https://www.openmpi.org) | The Open MPI Project is an open source Message Passing Interface implementation that is developed and maintained by a consortium of academic, research, and industry partners. 
**mpi** | <a id="py-mpi4py" class="soft_anch">`py-mpi4py`</a> | -&nbsp;`3.0.0_py27` | [Website](https://bitbucket.org/mpi4py/mpi4py) | MPI for Python provides Python bindings for the Message Passing Interface (MPI) standard. It is implemented on top of the MPI-1/2/3 specification and exposes an API which grounds on the standard MPI-2 C++ bindings. 
**networking** | <a id="gasnet" class="soft_anch">`gasnet`</a> | -&nbsp;`1.30.0` | [Website](https://gasnet.lbl.gov) | GASNet is a language-independent, low-level networking layer that provides network-independent, high-performance communication primitives tailored for implementing parallel global address space SPMD languages and libraries. 
**networking** | <a id="libfabric" class="soft_anch">`libfabric`</a> | -&nbsp;`1.6.0`<br/>-&nbsp;`1.6.2` | [Website](https://ofiwg.github.io/libfabric/) | The Open Fabrics Interfaces (OFI) is a framework focused on exporting fabric communication services to applications. Libfabric is the library that defines and exports the user-space API of OFI. 
**networking** | <a id="ucx" class="soft_anch">`ucx`</a> | -&nbsp;`1.2.1`<br/>-&nbsp;`1.4.0`<br/>-&nbsp;`1.3.1` | [Website](http://www.openucx.org) | UCX is a communication library implementing high-performance messaging for MPI/PGAS frameworks. 
**parser** | <a id="xerces-c" class="soft_anch">`xerces-c`</a> | -&nbsp;`3.2.1` | [Website](https://xerces.apache.org/xerces-c/index.html) | Xerces-C++ is a validating XML parser written in a portable subset of C++. 
